neuralop/layers/tests/test_local_fno_block.py

```diff
@@ -12,60 +13,59 @@ def test_LocalFNOBlock_resolution_scaling_factor():
-            3, 4, n_modes[:dim], n_layers=1, diff_layers=[True], resolution_scaling_factor=0.5, 
-            use_mlp=True, mlp_dropout=mlp_dropout, mlp_expansion=mlp_expansion, mlp_skip=mlp_skip)
+    # Downsample outputs
+    block = LocalFNOBlocks(
+        3, 4, n_modes[:n_dim], default_in_shape=tuple(size[:n_dim]), n_layers=1, diff_layers=[True], disco_layers=[(n_dim == 2)], resolution_scaling_factor=0.5, 
+        use_mlp=True, mlp_dropout=mlp_dropout, mlp_expansion=mlp_expansion, mlp_skip=mlp_skip)
 
-        x = torch.randn(2, 3, *size[:dim])
-        res = block(x)
-        assert(list(res.shape[2:]) == [m//2 for m in size[:dim]])
-        
-        # Upsample outputs
-        block = LocalFNOBlocks(
-            3, 4, n_modes[:dim], n_layers=1, diff_layers=[True], resolution_scaling_factor=2,
-            use_mlp=True, mlp_dropout=mlp_dropout, mlp_expansion=mlp_expansion, mlp_skip=mlp_skip)
+    x = torch.randn(2, 3, *size[:n_dim])
+    res = block(x)
+    assert(list(res.shape[2:]) == [m//2 for m in size[:n_dim]])
+    
+    # Upsample outputs
+    block = LocalFNOBlocks(
+        3, 4, n_modes[:n_dim], default_in_shape=tuple(size[:n_dim]), n_layers=1, diff_layers=[True], disco_layers=[(n_dim == 2)], resolution_scaling_factor=2,
+        use_mlp=True, mlp_dropout=mlp_dropout, mlp_expansion=mlp_expansion, mlp_skip=mlp_skip)
 
-        x = torch.randn(2, 3, *size[:dim])
-        res = block(x)
-        assert res.shape[1] == 4 # Check out channels
-        assert(list(res.shape[2:]) == [m*2 for m in size[:dim]])
+    x = torch.randn(2, 3, *size[:n_dim])
+    res = block(x)
+    assert res.shape[1] == 4 # Check out channels
+    assert(list(res.shape[2:]) == [m*2 for m in size[:n_dim]])
 
 @pytest.mark.parametrize('norm', 
                          ['instance_norm', 'ada_in', 'group_norm'])
-def test_LocalFNOBlock_norm(norm):
-    """Test SpectralConv with upsampled or downsampled outputs
+@pytest.mark.parametrize('n_dim', [1,2,3])
+def test_LocalFNOBlock_norm(norm, n_dim):
+    """Test LocalFNOBlock with normalization
     """
     modes = (8, 8, 8)
     size = [10]*3
     mlp_dropout=0
     mlp_expansion=0.5
     mlp_skip='linear'
-    dim = 2
     ada_in_features = 4
     block = LocalFNOBlocks(
-        3, 4, modes[:dim], n_layers=1, diff_layers=[True], use_mlp=True, norm=norm, ada_in_features=ada_in_features,
+        3, 4, modes[:n_dim], default_in_shape=tuple(size[:n_dim]), n_layers=1, diff_layers=[True], disco_layers=[(n_dim == 2)], use_mlp=True, norm=norm, ada_in_features=ada_in_features,
         mlp_dropout=mlp_dropout, mlp_expansion=mlp_expansion, mlp_skip=mlp_skip)
 
     if norm == 'ada_in':
         embedding = torch.randn(ada_in_features)
         block.set_ada_in_embeddings(embedding)
 
-    x = torch.randn(2, 3, *size[:dim])
+    x = torch.randn(2, 3, *size[:n_dim])
     res = block(x)
-    assert(list(res.shape[2:]) == size[:dim])
\ No newline at end of file
+    assert(list(res.shape[2:]) == size[:n_dim])
```

Comment:
@mliuschi could we potentially add a correctness test - e.g. there a simplistic case where we could init the conv with a known kernel and verify it matches a regular convolution?

Reply:
Hmm, that seems like something that might be better suited for testing the DISCO layer itself, right? This is just for merging a working DISCO layer with FNO and the differential kernels. What do you think?

Reply:
Yes, I agree, I didn't get time to review before it got merged. Would you be able to make another small PR to add this @mliuschi?

Reply:
This might pose a slight issue because of the radial basis that we have defined: it is not straightforward to directly parameterize a known kernel under our basis. 

I think Boris has done something similar in his plotting example though: he defined something similar to a traditional edge detector filter and applied it over a sample image. Would something like that address this?

Reply:
Yes exactly, just some correctness check - doesn't have to be the full general case but better than just checking the shape.